### PR TITLE
Feature: Add support for specifying a cstdlib other than glibc.

### PR DESCRIPTION
**Pull request by matte1**
We'd like to be able to specify other stdlibs besides glibc like musl. This adds an extra layer of hierachy to the repository so that the stdlib can be specified. A single example for aarch64 musl was added as an example.

### CODE REVIEW

File: setup_toolchains.py [hunk 1]
```diff
@@ -3,41 +3,61 @@
...
-            'tool_prefix': 'x86_64-buildroot-linux-gnu',
-        },
-        '2022.08-1': {
-            'sha256': '861c1e8ad0a66e4c28e7a1f8319d68080ab0ff8d16a765e65540f1957203a190',
-            'platform_arch': 'x86_64',
-            'tool_prefix': 'x86_64-buildroot-linux-gnu',
+        'gnu': {
+          '2021.11-5': {
+              'sha256': '6fe812add925493ea0841365f1fb7ca17fd9224bab61a731063f7f12f3a621b0',
+              'platform_arch': 'x86_64',
```
Comment (agoessling) [inline setup_toolchains.py:9]: Doesn't this need a 'cstdlib' entry?  How did this pass the tests?

File: setup_toolchains.py [hunk 2]
```diff
@@ -3,41 +3,61 @@
...
+              'tool_prefix': 'x86_64-buildroot-linux-gnu',
+          },
+          '2022.08-1': {
+              'sha256': '861c1e8ad0a66e4c28e7a1f8319d68080ab0ff8d16a765e65540f1957203a190',
+              'platform_arch': 'x86_64',
```
Comment (agoessling) [inline setup_toolchains.py:14]: same here

File: README.md [hunk 3]
```diff
@@ -34,12 +34,14 @@ load("@bazel_bootlin//toolchains:toolchains.bzl", "bootlin_toolchain_deps")
 
 bootlin_toolchain_deps(
     architecture = "x86-64-core-i7",
+    cstdlib = "gnu",
```
Comment (agoessling) [inline README.md:37]: I think you are mixing up `gnu` and `glibc`  `linux-gnu` describes the os environment I think right?  Maybe we remove references to `gnu` entirely?

**Review by agoessling — CHANGES_REQUESTED** [review]

File: setup_toolchains.py (hunk 1 above)
Comment (matte1) [reply setup_toolchains.py:9]: Its under a cstdlib right? The GNU above?

Comment (matte1) [reply setup_toolchains.py:9]: Done

File: setup_toolchains.py (hunk 2 above)
Comment (matte1) [reply setup_toolchains.py:14]: Done

File: README.md (hunk 3 above)
Comment (matte1) [reply README.md]: Done

**Review by agoessling — APPROVED** [review]

**Pull request merged**